src/ert/job_queue/job_queue_node.py

```diff
@@ -183,73 +215,86 @@ def runtime(self) -> float:
 
         return self._end_time - self._start_time
 
-    # pylint: disable=too-many-branches, too-many-statements
     def _job_monitor(
         self, driver: "Driver", pool_sema: Semaphore, max_submit: int
     ) -> None:
         submit_status = self.submit(driver)
         if submit_status is not JobSubmitStatusType.SUBMIT_OK:
-            self._set_status(JobStatusType.JOB_QUEUE_DONE)
+            self._set_queue_status(JobStatusType.JOB_QUEUE_DONE)
 
-        current_status = self.refresh_status(driver)
+        end_status = self._poll_until_done(driver)
+        self._handle_end_status(driver, pool_sema, end_status, max_submit)
 
+    def _poll_until_done(self, driver: Driver) -> JobStatusType:
+        current_status = self._poll_queue_status(driver)
+        backoff = _BackoffFunction()
         # in the following loop, we increase the sleep time between loop iterations as
         # long running realizations do not change state often, and too frequent querying
         # with many realizations starves other threads for resources.
-        initial_sleep_seconds = 1
-        time_sleep_seconds = initial_sleep_seconds
-        max_sleep_seconds = 30
-        time_until_longer_sleep_seconds = 30
-        use_random_sleep_offset = False
         while self.is_running(current_status):
             if (
                 self._start_time is None
                 and current_status == JobStatusType.JOB_QUEUE_RUNNING
             ):
                 self._start_time = time.time()
-            if (
-                self._start_time is not None
-                and time_sleep_seconds != max_sleep_seconds
-                and time.time() - self._start_time > time_until_longer_sleep_seconds
-            ):
-                time_sleep_seconds = max_sleep_seconds
-                use_random_sleep_offset = True
-            time.sleep(
-                time_sleep_seconds + use_random_sleep_offset * random.randint(-5, 5)
-            )
-            if self._max_runtime is not None and self.runtime >= self._max_runtime:
+            if self._start_time is not None:
+                elapsed_time = time.time() - self._start_time
+            else:
+                elapsed_time = 0.0
+            time.sleep(backoff(elapsed_time))
+            if self._exceeded_allowed_runtime():
                 self._kill(driver)
                 # We sometimes end up in a state where we are not able to kill it,
                 # so we end up flooding the logs with identical statements, so we
                 # check before we log.
```

Comment:
these comments need to be moved down to `_log_kill_timeout_status` if we want to keep them